services/device-registry-jdbc/src/test/java/org/eclipse/hono/deviceregistry/jdbc/impl/AbstractJdbcRegistryTest.java

```diff
@@ -76,7 +76,7 @@ enum DatabaseType {
     }
     protected static final Span SPAN = NoopSpan.INSTANCE;
 
-    private static final DatabaseType DEFAULT_DATABASE_TYPE = DatabaseType.H2;
+    private static final DatabaseType DEFAULT_DATABASE_TYPE = DatabaseType.POSTGRESQL;
```

Comment:
Yes, the SQL functions we use are only supported by POSTGRESQL. 
If I change to H2 I get the following error:
Function "JSONB_ARRAY_ELEMENTS" not found;

Reply:
Ok, what you are now doing is effectively disabling the running of the tests using an H2 database. Instead, you are now running the tests against a PostgreSQL 12 (the default container image name) and then against a PostgreSQL 14 instance (defined in the `device-registry-jdbc/pom.xml` file.

The overall idea with the JDBC based registry is that it will work with different types of (relational) DBs. The basic one being H2. The DEFAULT_DATABASE_TYPE is therefore set to H2 in order to run the tests against H2 when the surefire maven plugin executes without special configuration setting the database type explicitly.

So this property needs to remain set to value H2. Your problem seems to be that you have modified the `base` set of SQL statements which is being used as the baseline for all types of DBs. The base-jdbc package, however, also supports overriding the base set of SQL statements by means of DB type specific yaml files.

In your case, you should probably add your custom SQL statement definitions to the `base.postgresql.sql.yaml` file instead. However, in order to make sure that this also works with H2 you will need to add corresponding statements to `base.h2.sql.yaml` as well ...